### PR TITLE
add an option to disable required/optional strictness

### PR DESCRIPTION
**Pull request by blampe**
@breerly @kriskowal @junchaowu

### CODE REVIEW

**Comment by kriskowal** [comment]
:100:

File: CHANGELOG.rst [hunk 1]
```diff
@@ -4,7 +4,7 @@ Releases
 0.3.3 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Added an option to disable "required/optional" strictness on structs.
```
Comment (blampe) [inline CHANGELOG.rst]: @breerly I'm anticipating you'll tell me this belongs in 0.4.0 :)

I could also put "print as JSON" in there.

**Comment by breerly** [comment]
I think that the default should be disabled, since that's how Apache Thrift works.

File: thriftrw/loader.py [hunk 2]
```diff
@@ -33,17 +33,25 @@ class Loader(object):
...
             The protocol to use to serialize and deserialize types. Defaults
             to the binary protocol.
+
+        :param strict:
+            When ``True`` this enforces that "required" or "optional" always be
+            specified for struct elements. This is desirable because Apache
+            Thrift generates different behaviors for elements without
+            "required" or "optional" depending on the language. Setting this to
+            ``False`` is only recommended for compatibility with existing
+            Thrift files.
```
Comment (abhinav) [inline thriftrw/loader.py]: Mention the default value?

**Comment by abhinav** [comment]
@breerly I disagree. Strict should be on by default because always specifying requiredness is the preferred behavior, and we want to encourage that. Disabling it by default will result in even more reliance on the default requiredness.

**Comment by abhinav** [comment]
LGTM

**Comment by blampe** [comment]
dropping this in favor of 0.4 release

**Pull request closed without merging**